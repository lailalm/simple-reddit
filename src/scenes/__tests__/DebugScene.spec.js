/* eslint-disable react/jsx-closing-tag-location */
import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';

import ConnectedDebugScene, { DebugScene } from 'scenes/DebugScene';

import {
  generateTopic,
  clearTopic,
} from 'actions/DebugAction';

import {
  GENERATE_TOPIC,
  CLEAR_TOPIC,
} from 'utils/ActionTypes';


// Snapshot for CreateTopicScene React Component
describe('>>> DebugScene --- Snapshot', () => {
  it('+++ capturing Snapshot of DebugScene', () => {
    const renderedValue = renderer.create(<DebugScene />)
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('>>> DebugScene --- Shallow Render', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<DebugScene />);
  });

  it('+++ render the dummy component', () => {
    expect(wrapper.length).toEqual(1);
  });
});

describe('>>> CreateTopicScene --- React-Redux (Mount + wrapping in <Provider>)', () => {
  const origConsole = console.error;
  const initialState = {};
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    console.error = () => {};
    store = mockStore(initialState);
    wrapper = mount(<Provider store={store} >
      <ConnectedDebugScene />
    </Provider>);
  });

  // HACK: There isn't any react native adapter, so I'm using the react adapter.
  //       This will temporarily silence propType warnings
  afterEach(() => {
    console.error = origConsole;
  });

  it('+++ render the connected component', () => {
    expect(wrapper.find(ConnectedDebugScene).length).toEqual(1);
  });

  it('+++ check action on dispatching ', () => {
    store.dispatch(generateTopic(10));
    store.dispatch(clearTopic());
    const action = store.getActions();
    expect(action[0].type).toBe(GENERATE_TOPIC);
    expect(action[1].type).toBe(CLEAR_TOPIC);
  });
});
