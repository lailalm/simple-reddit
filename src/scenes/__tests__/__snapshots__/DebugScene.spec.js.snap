// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> DebugScene --- Snapshot +++ capturing Snapshot of DebugScene 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            Object {
              "borderRadius": 66.5,
              "paddingHorizontal": 18,
            },
            undefined,
            undefined,
            Object {
              "backgroundColor": "#51A3A3",
              "marginTop": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                Object {
                  "textAlign": "center",
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Add 10 Generated Topics
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            Object {
              "borderRadius": 66.5,
              "paddingHorizontal": 18,
            },
            undefined,
            undefined,
            Object {
              "backgroundColor": "#51A3A3",
              "marginTop": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                Object {
                  "textAlign": "center",
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Add 20 Generated Topics
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            Object {
              "borderRadius": 66.5,
              "paddingHorizontal": 18,
            },
            undefined,
            undefined,
            Object {
              "backgroundColor": "#DD7373",
              "marginTop": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                Object {
                  "textAlign": "center",
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Clear All Topic
        </Text>
      </View>
    </View>
  </View>
</View>
`;
